sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySampler.java

```diff
@@ -94,7 +96,18 @@ public SamplingResult shouldSample(
             .getOptions()
             .getInternalTracesSampler()
             .sample(new SamplingContext(transactionContext, null));
-    // TODO [POTEL] if sampling decision = false, we should record it in client report
+
+    if (!sentryDecision.getSampled()) {
+      scopes
+          .getOptions()
+          .getClientReportRecorder()
+          .recordLostEvent(DiscardReason.SAMPLE_RATE, DataCategory.Transaction);
+      scopes
+          .getOptions()
+          .getClientReportRecorder()
+          .recordLostEvent(DiscardReason.SAMPLE_RATE, DataCategory.Span);
```

Comment:
Is there always only one span here, or should we also count the children?

Reply:
sampler is invoked per span, before it's even started, we don't know the number of children it'll eventually have at this point. I can add something to the TODO list to see if we can come up with a number after the fact, but I'm not sure there's a callback that'll allow us to do so.

Reply:
Found a way to record for child spans as well. We now record a lost span when copying a parent sampling decision that is `false`.